grails-app/services/org/pih/warehouse/requisition/RequisitionIdentifierService.groovy

```diff
@@ -0,0 +1,33 @@
+package org.pih.warehouse.requisition
+
+import org.pih.warehouse.core.IdentifierService
+import org.pih.warehouse.core.identification.BlankIdentifierResolver
+import org.pih.warehouse.shipping.Shipment
+
+class RequisitionIdentifierService extends IdentifierService implements BlankIdentifierResolver<Requisition> {
+
+    @Override
+    String getPropertyKey() {
+        return "requisition"
+    }
+
+    @Override
+    protected Integer countDuplicates(String requestNumber) {
```

Comment:
for 1 and 2, yes can do. Thanks for the context on the why. Maybe we can add a dev task in the next release to refactor this properly, or at least a discovery to determine the work required. The nice thing about the ids being the same is that it's simpler to cross-reference them, but yes if we're refactoring to not need them, we could remove the check here on other tables.

Another option would be to not check for duplicates in other tables at all, and instead of simply assigning shipmentNumber=requestNumber, we do something like:
```
shipmentNumber = shipmentIdentifierService.generate(IdentifierGeneratorParams.builder()
                .formatOverride("\${custom.requestNumber}-\${random}")
                .customKeys(['requestNumber': requisition.requestNumber])
                .build())
```
So given requestNumber ABC123, shipmentNumber would be ABC123-123456.